### PR TITLE
SUS-2006: Implement rate limiting for chat

### PR DESCRIPTION
**Pull request by TK-999**
## Description
Prevent the user from sending any more messages for N seconds if they sent more than K messages in the last L seconds. The values of N, K, L are customizable via environment variables config.

## Ticket & other links
* http://wikia-inc.atlassian.net/browse/SUS-2006
* https://github.com/Wikia/app/pull/13054

### CODE REVIEW

File: test/userFreezer.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	beforeEach(function () {
+		fridge = new UserFreezer(10 /* messages per */, 1 /* second */, 5 /* seconds freeze time */);
+	});
+
+	it('applies freeze after the specified amount of messages', function () {
+		let elsa = new models.User(),
+			frozen,
+			elsaFreezeStart;
+
+		for (let i = 1; i <= 11; i++) {
```
Comment (sqreek) [inline test/userFreezer.js]: counting from 1?
Is it a Basic?
:trollface:

**Comment by sqreek** [comment]
FYI: We have rate limiting service that could be use in chat. But for this usage I think it's ok for this  implementation

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,16 @@
+dist: trusty
+sudo: false
 language: node_js
-node_js: ["0.12", "6"]
-services: ["redis"]
+node_js:
+ - "4"
+services: redis
```
Comment (macbre) [inline .travis.yml]: Do we want to change Travis env settings as a part of this fix?

Comment (TK-999) [inline .travis.yml]: It's already done in SUS-2191 so I will get rid of this change here :)

**Pull request merged**